Sources/SwiftProtobuf/Internal.swift

```diff
@@ -0,0 +1,42 @@
+// Sources/SwiftProtobuf/Message.swift - Message support
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Internal helpers on Messages for the library. These are public
+/// just so the generated code can call them, but shouldn't be called
+/// by developers directly.
+///
+// -----------------------------------------------------------------------------
+
+
+import Foundation
+import Swift
+
+public struct Internal {
```

Comment:
There will be some limits to what we can move—e.g., generic types can't be nested yet, so those would have to stay at the top level. But there's probably a good amount of public-as-implementation-detail stuff that we could put in there and it would be a good idea to do that.